lib/screens_web/controllers/screens_by_alert_controller.ex

```diff
@@ -3,20 +3,30 @@ defmodule ScreensWeb.ScreensByAlertController do
 
   alias Screens.ScreensByAlert
 
+  def index(conn, %{"ids" => ""}) do
+    json(conn, %{})
+  end
+
+  # `ids` must be a comma-separated list of integers.
+  # If any item in the list is an invalid integer, 400 is returned.
   def index(conn, %{"ids" => alert_ids_string}) do
-    if alert_ids_string == "" do
-      json(conn, %{})
-    else
-      screens_by_alerts =
-        alert_ids_string
-        |> String.split(",")
-        |> ScreensByAlert.get_screens_by_alert()
+    alert_ids =
+      alert_ids_string
+      |> String.trim()
+      |> String.split(",")
 
-      json(conn, screens_by_alerts)
+    if valid_alert_ids_param?(alert_ids) do
+      json(conn, ScreensByAlert.get_screens_by_alert(alert_ids))
+    else
+      Plug.Conn.send_resp(conn, 400, "Invalid alert ID found")
     end
   end
 
   def index(conn, _) do
     json(conn, %{})
   end
+
+  defp valid_alert_ids_param?(alert_ids) do
+    Enum.all?(alert_ids, &(Integer.parse(&1) != :error))
```

Comment:
While I don't see this popping up, I think it should be fine to allow a query param like this. As long as they are integers, it would be harmless.

Reply:
As long as strings with spaces don't end up getting through to the internal code, since e.g. `ScreensByAlert.get_screens_by_alert(["123   "])` would return nothing even if `"123"` is a valid alert ID

Reply:
Ahh, didn't think about that edge case. For that, we'll need a combination of your 1st and 3rd suggestions above. `Integer.parse/1` actually successfully parsed strings that are numbers with spaces mixed in. It just adds everything after the first space to the remainder. Matching on no remainder tackles this.